File: 4354-unique-middle-element/unique-middle-element.java
class Solution {
    public boolean isMiddleElementUnique(int[] nums) {
        int n=nums.length;
        int mid=nums[n/2];
        //System.out.println(mid);
        
        for(int i=0; i<n/2;i++){
            if(nums[i]==mid || nums[i+(n/2)+1]==mid){
                return false;
            }
        }
        return true;
    }
}

File: 4354-unique-middle-element/unique-middle-element.py
class Solution:
    def isMiddleElementUnique(self, nums: list[int]) -> bool:
        n=len(nums)
        freq=Counter(nums)
        mid=nums[n//2]
        for key,val in freq.items():
            if key==mid and val==1:
                return True
        return False